src/NeuralNet.cpp

```diff
@@ -25,6 +29,7 @@ namespace PyreNet {
     // Predictor
 
     std::vector<double> NeuralNet::predict(const std::vector<double> &input) {
+        if (this->layers.size() < 3 || this->inputSize == 0) throw InvalidNetworkSize();
```

Comment:
Since the input is not stored in the layers object, the layer has a minimum size of one. I think enforcing a middle layer is unnecessarily restrictive